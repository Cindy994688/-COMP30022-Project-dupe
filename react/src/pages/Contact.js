import React, { Component } from 'react';
import { Platform, StyleSheet, View, Image, Text } from 'react-native';
import './pages.css'
import './darkmode.css'
import './access.css'
import axios from 'axios'
import { makeStyles } from '@material-ui/core/styles';
import TextField from '@material-ui/core/TextField';


class Contact extends React.Component {
  constructor() {
      super();
      this.state = {
          email: "",
          name: "",
          message: ""
      };
  }

  render (props) {
      return (
        <div>
        <div className={"contact"+this.props.colourMode}>
        <br /><br />


          <div className = {"bioTextStable"+this.props.colourMode}>If you are interested in any of the
          projects or information you see on this page, please contact us
          below, or email us directly through one of the email addresses listed in our
          profiles. We look forward to hearing from you!  <p></p> </div>


          <form action="/email" id="contact-form" onSubmit={this.handleSubmit.bind(this)} method="POST">
              <div>
                  <div className = {"bioTextContact"+this.props.colourMode}>
                  <label htmlFor="name">Name</label>
                  </div>
                  <input type="text" className={"form-control"+this.props.colourMode+ " contactForm"+this.props.colourMode} placeholder = "Enter Name"
                  value={this.state.name} onChange={this.onNameChange.bind(this)}/>
              </div>
              <div>
                <div className = {"bioTextContact"+this.props.colourMode}>
                <label htmlFor="exampleInputEmail1">Email address</label>
                </div>
                <input type="email" className={"form-control"+this.props.colourMode} className={"contactForm"+this.props.colourMode} aria-describedby="emailHelp" placeholder = "Enter Email"
                value={this.state.email} onChange={this.onEmailChange.bind(this)}/>
              </div>
              <div>
                <div className = {"bioTextContact"+this.props.colourMode}>
                <label htmlFor="message">Message</label>
                </div>
                <div>
                <textarea className={"form-control"+this.props.colourMode} rows="5" className={"messageForm"+this.props.colourMode} placeholder = "Enter Message"
                value={this.state.message} onChange={this.onMessageChange.bind(this)}></textarea>
                </div>
              </div>
            <button type="submit" className={"btn"+this.props.colourMode+ " btn-primary"+this.props.colourMode+ " buttonsubmit"+this.props.colourMode}>Submit</button>
          </form>

        </div>
        </div>
      )
    }

    onNameChange(event) {
     this.setState({name: event.target.value});
   }

   onEmailChange(event) {
     this.setState({email: event.target.value});
   }

   onMessageChange(event) {
     this.setState({message: event.target.value});
   }

   handleSubmit(event) {
     //console.log(this.state.email);
     event.preventDefault();
     axios.post('/email', {
      name: this.state.name,
      email: this.state.email,
      message: this.state.message,
    });
    this.resetForm();
  }


  resetForm(){
    this.setState({name: '', email: '', message: ''})
  }
}



  const styles = StyleSheet.create({
    MainContainer: {
      flex: 1,
      justifyContent: 'center',
      alignItems: 'center',
      padding: "50px"
    },
    text: {
      marginTop: 30,
      fontSize: 40,
      color: '#43203A',
      fontFamily: ' "Kaushan", Times, serif',
      fontWeight: 'bold',
      alignItems: 'center',
      padding: "10px",
      display: 'inline-block',
      textAlign: 'center'
    },
  });

export default Contact;
