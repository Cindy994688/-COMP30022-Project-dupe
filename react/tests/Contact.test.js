/**
 * @jest-environment jsdom
 */

import React from 'react';
import renderer from 'react-test-renderer';
import Contact from '../src/pages/Contact';
const puppeteer = require('puppeteer');

test('it renders correctly', () => {
  component = renderer.create(<Contact/>)
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('E2E: Contact form can send messages', async() => {
   const browser = await puppeteer.launch({headless: false,});
   const page = await browser.newPage();
   await page.setViewport({width:1535, height:756});
   await page.goto('http:///localhost:3000/');
   await page.waitFor('#contact-form');
   const name= await page.waitForXPath("//input[1]");
   await name.type("testing");
   const email= await page.waitForXPath("//input[@type='email']");
   await email.type("test@PPP.com");
   const message= await page.waitForXPath("//textarea");
   await message.type("Regular testing the contacts page with puppeteer");
   await page.waitFor(2000);
   const button = await page.waitForXPath("//button");
   await button.click();
   await page.waitFor(2000);
   await browser.close();
 }, 100000);
