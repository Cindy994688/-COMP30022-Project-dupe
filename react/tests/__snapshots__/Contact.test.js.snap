// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<div>
  <div
    className="contactundefined"
  >
    <br />
    <br />
    <div
      className="bioTextStableundefined"
    >
      If you are interested in any of the projects or information you see on this page, please contact us below, or email us directly through one of the email addresses listed in our profiles. We look forward to hearing from you!  
      <p />
       
    </div>
    <form
      action="/email"
      id="contact-form"
      method="POST"
      onSubmit={[Function]}
    >
      <div>
        <div
          className="bioTextContactundefined"
        >
          <label
            htmlFor="name"
          >
            Name
          </label>
        </div>
        <input
          className="form-controlundefined contactFormundefined"
          onChange={[Function]}
          placeholder="Enter Name"
          type="text"
          value=""
        />
      </div>
      <div>
        <div
          className="bioTextContactundefined"
        >
          <label
            htmlFor="exampleInputEmail1"
          >
            Email address
          </label>
        </div>
        <input
          aria-describedby="emailHelp"
          className="contactFormundefined"
          onChange={[Function]}
          placeholder="Enter Email"
          type="email"
          value=""
        />
      </div>
      <div>
        <div
          className="bioTextContactundefined"
        >
          <label
            htmlFor="message"
          >
            Message
          </label>
        </div>
        <div>
          <textarea
            className="messageFormundefined"
            onChange={[Function]}
            placeholder="Enter Message"
            rows="5"
            value=""
          />
        </div>
      </div>
      <button
        className="btnundefined btn-primaryundefined buttonsubmitundefined"
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</div>
`;
